core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -42,18 +60,43 @@ private[spark] case class AccumulatorMetadata(
  * `OUT` should be a type that can be read atomically (e.g., Int, Long), or thread-safely
  * (e.g., synchronized collections) because it will be read from other threads.
  */
-abstract class AccumulatorV2[IN, OUT] extends Serializable {
+abstract class AccumulatorV2[@specialized(Int, Long, Double) IN, OUT] extends Serializable {
```

Comment:
So we could do that, but would the specialized annotation maybe easier to maintain?